src/item.cpp

```diff
@@ -1090,13 +1090,32 @@ bool item::merge_charges( const item &rhs )
 ret_val<bool> item::put_in( const item &payload, item_pocket::pocket_type pk_type )
 {
     ret_val<bool> result = contents.insert_item( payload, pk_type );
+    if( !result.success() ) {
+        debugmsg( "tried to put an item (%s) count (%d) in a container (%s) that cannot contain it: %s",
+                  payload.typeId().str(), payload.count(), typeId().str(), result.str() );
+    }
     if( pk_type == item_pocket::pocket_type::MOD ) {
         update_modified_pockets();
     }
     on_contents_changed();
     return result;
 }
 
+ret_val<bool> item::put_in( const item &payload )
```

Comment:
There is a very similar , but not identical `item::migrate_content_item` function which should have the same logic as `item::put_in` - we should update it probably to get rid of duplicate code (not necessary in current PR).

Reply:
i don't think this function should exist at all, because picking an implicit pocket is very dangerous.

Reply:
> i don't think this function should exist at all, because picking an implicit pocket is very dangerous.

@KorGgenT  `item::migrate_content_item` or `item::put_in`?
`item::put_in` w/o `pk_type` is safe as long as there are no rifles with a backpack that stores spare mods or magazines. For spawning such items (with many pocket types for one item type), really need to change itemgroup to specify a specific pocket or pocket type. But this PR is just a fix. It makes the game work, not improves it. Let improving itemgroup in another PR.

Reply:
If this is a bandaid bugfix it shouldn't be in the item interface, make it a helper function local to item_group that guesses the pocket type and only use it there, that leaves less code to rip back out later and doesn't risk other people using it.

Reply:
To clarify, invoking it should look like
```
const item_pocket::pocket_type pk_type = guess_pocket_for( container, new_item );
container.put_in( new_item, pk_type );
```

Reply:
Okay, I agree.

Reply:
Done.